src/decays/experimental_constraints.cpp

```diff
@@ -0,0 +1,579 @@
+// ====================================================================
+// This file is part of FlexibleSUSY.
+//
+// FlexibleSUSY is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// FlexibleSUSY is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with FlexibleSUSY.  If not, see
+// <http://www.gnu.org/licenses/>.
+// ====================================================================
+
+
+/**
+ * @file HiggsTools_interface.cpp
+ *
+ * @brief contains interface to HiggsTools
+ */
+
+#include "experimental_constraints.hpp"
+#include "decays/standard_model_decays.hpp"
+
+#include <gsl/gsl_min.h>
+#include <gsl/gsl_errno.h>
+#include <boost/math/distributions/chi_squared.hpp>
+
+#include <algorithm>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <functional>
+#include <random>
+
+#ifdef ENABLE_HIGGSTOOLS
+#include "Higgs/Predictions.hpp"
+#include "Higgs/Bounds.hpp"
+#include "Higgs/Signals.hpp"
+namespace HP = Higgs::predictions;
+#endif
+
+#ifdef ENABLE_LILITH
+#include <Python.h>
+#include "lilith.h"
+#include "lilith.c"
+#endif
+
+namespace flexiblesusy {
+
+namespace {
+
+// Whether to calculate the X->gaga and X->gg decay widths in terms of the
+// effective tree-level couplings (calcHgamgam=calcggH=true) or by rescaling
+// the X->gaga and X->gg decay by the squared of the effective gamgam and gg
+// couplings (calcHgamgam=calcggH=false). True would make sense only if we
+// would not compute those loop-induced decays ourself.
+constexpr bool calcggH     = false;
+constexpr bool calcHgamgam = false;
+
+// relative BSM Higgs-like state mass uncertainty
+// example: 0.03 means 3% uncertainty
+constexpr double relMassError = 0.03;
+
+#ifdef ENABLE_HIGGSTOOLS
+// Ref. model for computing brs and xsections on the HiggsTools side
+constexpr auto refModel = HP::ReferenceModel::SMHiggsInterp;
+
+double minChi2SM_hs(const double mhSM, std::string const& higgssignals_dataset) {
+   const auto signals = Higgs::Signals {higgssignals_dataset};
+
+   auto pred = Higgs::Predictions();
+   auto& s = pred.addParticle(HP::BsmParticle("hSM", HP::ECharge::neutral, HP::CP::even));
+   auto effc = HP::scaledSMlikeEffCouplings(1.0);
+   s.setMass(mhSM);
+   s.setMassUnc(0.);
+   effectiveCouplingInput(
+      s, effc,
+      refModel,
+      calcggH, calcHgamgam
+   );
+   return signals(pred);
+}
+
+void print_effc(double mass, HP::NeutralEffectiveCouplings const& effC) {
+   std::cout << "Effective couplings for particle of mass " << mass << '\n';
```

Comment:
Please do not end every `cout` statement with `std::endl`, because this is very slow. Please use `'\n'` instead. It would be ok to `std::endl` in the very last line, though.

Reason: `std::endl` calls `coud.flush()`, which is quite expensive and which is not needed at the end of every line here.

Reply:
This is not needed. This was just some dubug helper function. I've removed it in 9166aaaae3b6d65ce5464f5c991a0ec16078a238.